submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestReleaser.kt

```diff
@@ -1,37 +1,57 @@
 
 import ac.uk.ebi.biostd.persistence.common.model.RequestStatus.PROCESSED
+import ac.uk.ebi.biostd.persistence.common.model.SubmissionRequest
 import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
 import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceService
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestFilesPersistenceService
 import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestPersistenceService
 import ac.uk.ebi.biostd.persistence.filesystem.api.FileStorageService
 import ebi.ac.uk.extended.model.ExtFile
 import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.extended.model.FireFile
 import mu.KotlinLogging
 import uk.ac.ebi.events.service.EventsPublisherService
 import uk.ac.ebi.extended.serialization.service.ExtSerializationService
 import uk.ac.ebi.extended.serialization.service.fileSequence
 
 private val logger = KotlinLogging.logger {}
 
+@Suppress("LongParameterList")
 class SubmissionRequestReleaser(
     private val fileStorageService: FileStorageService,
     private val serializationService: ExtSerializationService,
     private val eventsPublisherService: EventsPublisherService,
     private val queryService: SubmissionPersistenceQueryService,
     private val persistenceService: SubmissionPersistenceService,
     private val requestService: SubmissionRequestPersistenceService,
+    private val filesRequestService: SubmissionRequestFilesPersistenceService,
 ) {
+
     /**
      * Check the release status of the submission and release it if released flag is true.
      */
     fun checkReleased(accNo: String, version: Int): ExtSubmission {
         val request = requestService.getFilesCopiedRequest(accNo, version)
-        val sub = queryService.getExtByAccNoAndVersion(accNo, version, includeFileListFiles = true)
-        if (sub.released) releaseSubmission(sub)
-        requestService.updateRequestStatus(sub.accNo, sub.version, PROCESSED)
+        if (request.submission.released) releaseRequest(request)
+        requestService.updateRequestStatus(request.submission.accNo, request.submission.version, PROCESSED)
         eventsPublisherService.submissionSubmitted(accNo, request.notifyTo)
-        return sub
+        return request.submission
+    }
+
+    private fun releaseRequest(
+        request: SubmissionRequest,
+    ) {
+        val sub = request.submission
+        filesRequestService
+            .getSubmissionRequestFiles(sub.accNo, sub.version, request.currentIndex)
+            .filterNot { it.file is FireFile && (it.file as FireFile).published }
+            .forEach {
+                fileStorageService.releaseSubmissionFile(it.file, sub.relPath, sub.storageMode)
+                filesRequestService.saveSubmissionRequestFile(it)
+                requestService.updateRequestIndex(sub.accNo, sub.version, it.index)
+            }
+        persistenceService.setAsReleased(sub.accNo)
```

Comment:
This call is not needed in this case since it will only go here if the submission in the request is already released so, in theory, it should never go here if the released field is false